src/Aspire.Dashboard/DashboardWebApplication.cs

```diff
@@ -284,6 +294,27 @@ public DashboardWebApplication(
             .Select(c => c.Name)
             .ToArray();
 
+        // TODO: Revisit whether this is needed. It provides the ability to use a path base when forwarded headers don't set it,
+        //       e.g. when not using a reverse-proxy but instead just configuring an app host project to use a path base.
+        if (dashboardOptions.PathBase is not null)
+        {
+            _app.UsePathBase(dashboardOptions.PathBase);
+
+            if (_app.Environment.IsDevelopment())
+            {
+                // In development we want to ensure that requests outside of the path base return a 404 to mimic what would
+                // happen when the dashboard is being served from behind a reverse-proxy at a configured path. Note we still let
+                // ~/login requests through as that's what VS launches the browser to. In that case we redirect instead.
```

Comment:
Need intergration tests of this:

Base path + good path = 200
Base path + bad path = 404
Base path + /login = 302